pkg/time/fakeclock/fake_gettimeofday.c

```diff
@@ -24,36 +24,37 @@ extern int64_t TV_SEC_DELTA;
 extern int64_t TV_NSEC_DELTA;
 
 #if defined(__amd64__)
-inline int real_gettimeofday(struct timeval *tv, struct timezone *__tz)
+inline int real_gettimeofday(struct timeval *tv, struct timezone *tz)
 {
     int ret;
     asm volatile(
         "syscall"
         : "=a"(ret)
-        : "0"(__NR_gettimeofday), "D"(tv), "S"(__tz)
+        : "0"(__NR_gettimeofday), "D"(tv), "S"(tz)
         : "memory");
 
     return ret;
 }
 
 #elif defined(__aarch64__)
-inline int real_gettimeofday(struct timeval *tv)
+inline int real_gettimeofday(struct timeval *tv, struct timezone *tz)
 {
     register struct timeval *x0 __asm__("x0") = tv;
```

Comment:
```
./pkg/time/fakeclock/fake_gettimeofday.c:50:12: warning: returning 'struct timeval *' from a function with return type 'int' makes integer from pointer without a cast [-Wint-conversion]
     return x0;
            ^~
```
I think the type of `x0` should be fixed too :)

Reply:
The problem is that the `x0` is both the first argument and the return value. Maybe it should be converted to `int` after calling the system call 🤔 .

Reply:
It would give another warning `warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]` when I trying to convert it to `int`. What should I do? How about introducing another new variable `x2` (as using the same register) for receiving return value. I am NOT sure about using `x2` in syscall would break something convention.

Reply:
PTAL @YangKeao